### PR TITLE
Add `highlightChecked` prop to the `ShippingRatesControlPackage` in WC Blocks integration

### PR DESCRIPTION
**Pull request by opr**
This PR is part of https://github.com/woocommerce/woocommerce/pull/46150 and should be merged in time for WC 8.9. Ideally WC Subscriptions will ship this patch before WC 8.9 is released.

## Description

This PR adds new props to the `ShippingRatesControlPackage` used to display the recurring shipping packages. The purpose of them are to allow the selected method to be highlighted in the Checkout block.

`context` is whether the component is rendered in the Cart or Checkout block and `highlightChecked` determines if the checked option should be highlighted.

This change needs to be made here as well as in WC Core because we default to _not_ highlighting the selected item to preserve backward compatibility with any other consumers of the component. 

## How to test this PR

0. Be using https://github.com/woocommerce/woocommerce/pull/46150 or latest WC Core trunk if that is merged already.
1. Create 3 subscription items with different lengths, (weekly, yearly, monthly)
2. Add them to your cart and go to the Cart block.
3. Ensure the shipping options in the sidebar show correctly, i.e. without borders around each rate.

image

4. Go to the Checkout block. Ensure you can see the additional packages, and that they _do_ have borders around each rate

image

5. Switch back to WC Core trunk and run the tests again, the borders won't show but please ensure the shipping selectors for recurring packages still work and can change the method.

## Product impact

- [x] Added changelog entry (or does not apply)
- [x] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref
- [ ]  Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)

### CODE REVIEW

**Review by james-allan — APPROVED** [review]
Thanks for the PR @opr. This looks good to me. I've confirmed that the borders don't show on `trunk` of subscriptions core, and that's fixed by this branch.

| `trunk` | This branch |
|--------|--------|
| <p align="center">Screenshot 2024-04-05 at 10 58 01 am</br><sup><strong>Note:</strong> The board is missing from subscription shipping method.</sup></p>|Screenshot 2024-04-05 at 11 07 35 am| 

As you mentioned, I also confirmed that using WooCommerce 8.7 (current public version), the borders aren't shown at all (screenshot below) making this PR safe to deploy pre WC 8.9.0. 

<p align="center">
Screenshot 2024-04-05 at 11 20 48 am
</p>

I've also confirmed that the block cart isn't affected at all and that selecting different shipping methods are reflected in the cart totals etc. 

Please feel free to merge when appropriate. I'm not sure if you want to wait until it's merged on the WC side. I can also merge on your behalf if you'd like. :)

Just as an FYI, Woo Subscriptions has a release scheduled next week on April 11. This would be deployed as part of that version if merged before hand.

Thanks again.

**Comment by opr** [comment]
@james-allan thanks a lot for the review, and for the confirmation and screenshots of each step, that's great.

I would prefer to get the WC Blocks PR merged first (just so we know that the approach is correct, and the `highlightChecked` prop _will_ be added to the component).

When is the next release after April 11?

Thanks!

**Comment by james-allan** [comment]
> When is the next release after April 11?

We release on the second Thursday of each month so May 9th will be the next scheduled release.

**Comment by opr** [comment]
That's fantastic, thanks. WC 8.9 is slated for release on May 9th, so this works OK. I will see if we can't get it merged before 11th April though.

**Pull request merged**
